server/datastore/mysql/software.go

```diff
@@ -514,8 +516,8 @@ func loadCVEsBySoftware(
 	return cvesBySoftware, nil
 }
 
-func (ds *Datastore) LoadHostSoftware(ctx context.Context, host *fleet.Host) error {
-	software, err := listSoftwareDB(ctx, ds.reader, &host.ID, fleet.SoftwareListOptions{})
+func (ds *Datastore) LoadHostSoftware(ctx context.Context, host *fleet.Host, opts fleet.SoftwareListOptions) error {
```

Comment:
This comment is not to be tackled on this PR. Just commenting for small future improvements.

LoadHostSoftware should eventually be changed to:
```go
func (ds *Datastore) LoadHostSoftware(ctx context.Context, hostID uint, opts fleet.SoftwareListOptions) ([]fleet.Software, error) {
  // ...
}
```
A bit more cleaner than the current approach. (Otherwise you have to look at the implementation to know what's being set in the provided `host *fleet.Host`)